src/v/redpanda/application.cc

```diff
@@ -565,10 +565,11 @@ void application::wire_up_services() {
         wire_up_redpanda_services();
     }
     if (_proxy_config) {
-        construct_service(_proxy_client, to_yaml(*_proxy_client_config)).get();
+        construct_service(_proxy_client, to_yaml(*_proxy_client_config,
+          /*redact_secrets*/false)).get();
         construct_service(
           _proxy,
-          to_yaml(*_proxy_config),
+          to_yaml(*_proxy_config, /*redact_secrets*/false),
```

Comment:
instead of leaving a comment for each parameter, you can use a boolean strong type:

```c++
using redact_secrets = ss::bool_class<class redact_secrets_tag>;
...
foo(redact_secrets::no);
```

Reply:
Done